transcendence-app/src/chat/chat/infrastructure/db/postgres/chat-message.postgres.repository.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@nestjs/common';
+
+import { table } from '../../../../../shared/db/models';
+import { PostgresPool } from '../../../../../shared/db/postgres/postgresConnection.provider';
+import { BasePostgresRepository } from '../../../../../shared/db/postgres/postgres.repository';
+import { ChatMessageEntity } from '../chat-message.entity';
+import { IChatMessageRepository } from '../chat-message.repository';
+import { ChatMessagePaginationQueryDto } from '../../../../dto/chat-message.pagination.dto';
+import { ChatMessageQueryDto } from '../../../../dto/chat-message.dto';
+import { makeQuery } from '../../../../../shared/db/postgres/utils';
+
+@Injectable()
+export class ChatMessagePostgresRepository
+  extends BasePostgresRepository<ChatMessageEntity>
+  implements IChatMessageRepository
+{
+  constructor(protected pool: PostgresPool) {
+    super(pool, table.CHAT_MESSAGE);
+  }
+
+  async addMessageSet(
+    queryDto: Required<ChatMessageQueryDto>,
+  ): Promise<ChatMessageEntity | null> {
+    const { authorId, content, user1Id, user2Id } = queryDto;
+
+    const message = await makeQuery<ChatMessageEntity>(this.pool, {
+      text: `INSERT INTO ${this.table}("authorId", "content", "user1Id", "user2Id")
+      VALUES ($1, $2, LEAST($3, $4), GREATEST($3, $4))
```

Comment:
With LEAST and GREATEST we ensure to store users always in same order

Reply:
I would rather propose `senderId` and `receiverId` or `senderId` and `recipientId`

Reply:
Agree, I'll go with senderId and recipientId